hail/src/main/scala/is/hail/expr/ir/TableIR.scala

```diff
@@ -1261,6 +1260,16 @@ case class TableOrderBy(child: TableIR, sortFields: IndexedSeq[SortField]) exten
   protected[ir] override def execute(hc: HailContext): TableValue = {
     val prev = child.execute(hc)
 
+    if (sortFields.isEmpty)
+      return prev
+
+    if (sortFields.length <= prev.typ.key.length &&
```

Comment:
actually, this should look at the physical key, to be even stronger.

Reply:
Oops, I had intended to do that, good catch.